mr/src/main/java/org/apache/iceberg/mr/mapred/TableResolver.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Properties;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.hadoop.HadoopCatalog;
+import org.apache.iceberg.hadoop.HadoopTables;
+import org.apache.iceberg.mr.InputFormatConfig;
+
+final class TableResolver {
+
+  private TableResolver() {
+  }
+
+  static Table resolveTableFromJob(JobConf conf) throws IOException {
+    Properties properties = new Properties();
+    properties.setProperty(InputFormatConfig.CATALOG_NAME, extractProperty(conf, InputFormatConfig.CATALOG_NAME));
+    if (conf.get(InputFormatConfig.CATALOG_NAME).equals(InputFormatConfig.HADOOP_CATALOG)) {
+      properties.setProperty(InputFormatConfig.SNAPSHOT_TABLE, conf.get(InputFormatConfig.SNAPSHOT_TABLE, "true"));
+    }
+    properties.setProperty(InputFormatConfig.TABLE_LOCATION, extractProperty(conf, InputFormatConfig.TABLE_LOCATION));
+    properties.setProperty(InputFormatConfig.TABLE_NAME, extractProperty(conf, InputFormatConfig.TABLE_NAME));
+    return resolveTableFromConfiguration(conf, properties);
+  }
+
+  static Table resolveTableFromConfiguration(Configuration conf, Properties properties) throws IOException {
+    String catalogName = properties.getProperty(InputFormatConfig.CATALOG_NAME);
+    URI tableLocation = pathAsURI(properties.getProperty(InputFormatConfig.TABLE_LOCATION));
+    if (catalogName == null) {
+      throw new IllegalArgumentException("Catalog property: 'iceberg.catalog' not set in JobConf");
+    }
+    switch (catalogName) {
+      case InputFormatConfig.HADOOP_TABLES:
+        HadoopTables tables = new HadoopTables(conf);
+        return tables.load(tableLocation.getPath());
+      case InputFormatConfig.HADOOP_CATALOG:
+        String tableName = properties.getProperty(InputFormatConfig.TABLE_NAME);
+        TableIdentifier id = TableIdentifier.parse(tableName);
+        if (tableName.endsWith(InputFormatConfig.SNAPSHOT_TABLE_SUFFIX)) {
+          if (!Boolean.parseBoolean(properties.getProperty(InputFormatConfig.SNAPSHOT_TABLE,
+                  Boolean.TRUE.toString()))) {
+            String tablePath = id.toString().replaceAll("\\.", "/");
+            URI warehouseLocation = pathAsURI(tableLocation.getPath().replaceAll(tablePath, ""));
+            HadoopCatalog catalog = new HadoopCatalog(conf, warehouseLocation.getPath());
+            return catalog.loadTable(id);
+          } else {
+            return resolveMetadataTable(conf, tableLocation.getPath(), tableName);
+          }
+        } else {
+          URI warehouseLocation = pathAsURI(extractWarehousePath(tableLocation.getPath(), tableName));
+          HadoopCatalog catalog = new HadoopCatalog(conf, warehouseLocation.getPath());
+          return catalog.loadTable(id);
+        }
+      case InputFormatConfig.HIVE_CATALOG:
+        //TODO Implement HiveCatalog
```

Comment:
Hive adds a property to the conf called `name` which is whatever you've defined in the `CREATE TABLE name` part, so that property will include db and table name

Reply:
Should this throw an exception instead of returning null?